From 2eb2a661d7b75cab0ad65a16663a448f6d67320b Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Sun, 10 Aug 2014 18:20:12 -0400
Subject: [iommu] intel: correct ICS register offset

Message-id: <20140810182011.24055.77969.stgit@gir.stowe>
Patchwork-id: 87578
O-Subject: [RHEL7.1 PATCH 07/19] x86/iommu: correct ICS register offset
Bugzilla: 1127467
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Dutile <ddutile@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1127467
Upstream Status: 82aeef0bf03684b377678c00c05e613f30dca39c

commit 82aeef0bf03684b377678c00c05e613f30dca39c
Author: Li, Zhen-Hua <zhen-hual@hp.com>
Date:   Fri Sep 13 14:27:32 2013 +0800

    x86/iommu: correct ICS register offset

    According to Intel Vt-D specs, the offset of Invalidation complete
    status register should be 0x9C, not 0x98.

    See Intel's VT-d spec, Revision 1.3, Chapter 10.4, Page 98;

    Signed-off-by: Li, Zhen-Hua <zhen-hual@hp.com>
    Signed-off-by: Joerg Roedel <joro@8bytes.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/linux/intel-iommu.h b/include/linux/intel-iommu.h
index 78e2ada..d380c5e 100644
--- a/include/linux/intel-iommu.h
+++ b/include/linux/intel-iommu.h
@@ -55,7 +55,7 @@
 #define DMAR_IQT_REG 0x88 /* Invalidation queue tail register */
 #define DMAR_IQ_SHIFT 4 /* Invalidation queue head/tail shift */
 #define DMAR_IQA_REG 0x90 /* Invalidation queue addr register */
-#define DMAR_ICS_REG 0x98 /* Invalidation complete status register */
+#define DMAR_ICS_REG 0x9c /* Invalidation complete status register */
 #define DMAR_IRTA_REG 0xb8    /* Interrupt remapping table addr register */
 
 #define OFFSET_STRIDE  (9)
-- 
1.7.1